lightning-invoice/src/utils.rs

```diff
@@ -512,8 +512,9 @@ fn _create_invoice_from_channelmanager_and_duration_since_epoch_with_payment_has
 /// * Always select the channel with the highest inbound capacity per counterparty node
 /// * Prefer channels with capacity at least `min_inbound_capacity_msat` and where the channel
 ///   `is_usable` (i.e. the peer is connected).
-/// * If any public channel exists, the returned `RouteHint`s will be empty, and the sender will
-///   need to find the path by looking at the public channels instead
+/// * If any public channel exists, only public [`RouteHint`]s will be returned,
+/// * If any public, announced channel exists, no [`RouteHint`]s will be returned, as the sender
```

Comment:
Very nit picky:
We can have an announced channel with 6 confirmations and still include the [`RouteHint`] for that channel, no? Might specifically be worth mentioning 7+ confirmations here.

Though feel free to ignore this suggestion if you think this is too nit-picky / bring more confusion than clarification :).